drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/editor/ScenarioSimulationEditorKogitoWrapper.java

```diff
@@ -0,0 +1,320 @@
+        this.promises = promises;
+        this.dmnTypeService = dmnTypeService;
+        this.kogitoOracle = kogitoOracle;
+    }
+
+    @Override
+    protected void buildMenuBar() {
+        setMenus(fileMenuBuilder.build());
+        getMenus().getItems().forEach(menuItem -> menuItem.setEnabled(true));
+    }
+
+    @Override
+    public Promise<String> getContent() {
+        return transform(scenarioSimulationEditorPresenter.getModel());
+    }
+
+    @Override
+    public void setContent(String path, String value) {
+        unmarshallContent(value);
+    }
+
+    @Override
+    public void wrappedRegisterDock(String id, IsWidget widget) {
+        //
+    }
+
+    @Override
+    public void onImport(String fileContents, RemoteCallback<AbstractScesimModel> importCallBack, ErrorCallback<Object> importErrorCallback, AbstractScesimModel<? extends AbstractScesimData> scesimModel) {
+        //
+    }
+
+    @Override
+    public void onExportToCsv(RemoteCallback<Object> exportCallBack, ScenarioSimulationHasBusyIndicatorDefaultErrorCallback scenarioSimulationHasBusyIndicatorDefaultErrorCallback, AbstractScesimModel<? extends AbstractScesimData> scesimModel) {
+        //
+    }
+
+    @Override
+    public void onDownloadReportToCsv(RemoteCallback<Object> exportCallBack, ScenarioSimulationHasBusyIndicatorDefaultErrorCallback scenarioSimulationHasBusyIndicatorDefaultErrorCallback, AuditLog auditLog) {
+        //
+    }
+
+    @Override
+    public void validate(Simulation simulation, Settings settings, RemoteCallback<?> callback) {
+        //
+    }
+
+    /**
+     * This method is called when the main grid tab (Model) is focused
+     */
+    @Override
+    public void onEditTabSelected() {
+        super.onEditTabSelected();
+        scenarioSimulationEditorPresenter.onEditTabSelected();
+    }
+
+    /**
+     * This method adds specifically the Background grid and its related onFocus behavior
+     * @param scenarioGridWidget
+     */
+    @Override
+    public void addBackgroundPage(final ScenarioGridWidget scenarioGridWidget) {
+        addPage(new PageImpl(scenarioGridWidget, ScenarioSimulationEditorConstants.INSTANCE.backgroundTabTitle()) {
+            @Override
+            public void onFocus() {
+                super.onFocus();
+                onBackgroundTabSelected();
+            }
+        });
+    }
+
+    @Override
+    public void selectSimulationTab() {
+        final TabListItem item = (TabListItem) ((MultiPageEditorViewImpl) getWidget().getMultiPage().getView()).getTabBar().getWidget(SIMULATION_TAB_INDEX);
+        if (item != null) {
+            item.showTab(false);
+        }
+    }
+
+    @Override
+    public void selectBackgroundTab() {
+        final TabListItem item = (TabListItem) ((MultiPageEditorViewImpl) getWidget().getMultiPage().getView()).getTabBar().getWidget(BACKGROUND_TAB_INDEX);
+        if (item != null) {
+            item.showTab(false);
+        }
+    }
```

Comment:
We should avoid hardcoding the tab indexing. 
In past it caused troubles like: https://issues.redhat.com/browse/AF-2418

As part of fix for AF-2418 an API for dynamical index computation was introduced.

I created task for me https://issues.redhat.com/browse/DROOLS-5104

Reply:
@jomarko Fine for me, good catch! Please add me a reviewer when you completed the task.